libcudacxx/include/cuda/std/detail/libcxx/include/__ranges/concepts.h

```diff
@@ -0,0 +1,348 @@
+    __view_,
+    requires()(
+      requires(range<_Tp>),
+      requires(movable<_Tp>),
+      requires(enable_view<_Tp>)
+    ));
+
+  template <class _Tp>
+  _LIBCUDACXX_CONCEPT view = _LIBCUDACXX_FRAGMENT(__view_, _Tp);
+
+  template <class _Range>
+  _LIBCUDACXX_CONCEPT_FRAGMENT(
+    __simple_view_,
+    requires()(
+      requires(view<_Range>),
+      requires(range<const _Range>),
+      requires(same_as<iterator_t<_Range>, iterator_t<const _Range>>),
+      requires(same_as<sentinel_t<_Range>, sentinel_t<const _Range>>)
+    ));
+
+  template <class _Range>
+  _LIBCUDACXX_CONCEPT __simple_view = _LIBCUDACXX_FRAGMENT(__simple_view_, _Range);
+
+  // [range.refinements], other range refinements
+  template <class _Rp, class _Tp>
+  _LIBCUDACXX_CONCEPT_FRAGMENT(
+    __output_range_,
+    requires()(
+      requires(range<_Rp>),
+      requires(output_iterator<iterator_t<_Rp>, _Tp>)
+    ));
+
+  template <class _Rp, class _Tp>
+  _LIBCUDACXX_CONCEPT output_range = _LIBCUDACXX_FRAGMENT(__output_range_, _Rp, _Tp);
+
+  template <class _Tp>
+  _LIBCUDACXX_CONCEPT_FRAGMENT(
+    __forward_range_,
+    requires()(
+      requires(input_range<_Tp>),
+      requires(forward_iterator<iterator_t<_Tp>>)
+    ));
+
+  template <class _Tp>
+  _LIBCUDACXX_CONCEPT forward_range = _LIBCUDACXX_FRAGMENT(__forward_range_, _Tp);
+
+  template <class _Tp>
+  _LIBCUDACXX_CONCEPT_FRAGMENT(
+    __bidirectional_range_,
+    requires()(
+      requires(forward_range<_Tp>),
+      requires(bidirectional_iterator<iterator_t<_Tp>>)
+    ));
+
+  template <class _Tp>
+  _LIBCUDACXX_CONCEPT bidirectional_range = _LIBCUDACXX_FRAGMENT(__bidirectional_range_, _Tp);
+
+  template <class _Tp>
+  _LIBCUDACXX_CONCEPT_FRAGMENT(
+    __random_access_range_,
+    requires()(
+      requires(bidirectional_range<_Tp>),
+      requires(random_access_iterator<iterator_t<_Tp>>)
+    ));
+
+  template <class _Tp>
+  _LIBCUDACXX_CONCEPT random_access_range = _LIBCUDACXX_FRAGMENT(__random_access_range_, _Tp);
+
+  template <class _Tp>
+  _LIBCUDACXX_CONCEPT_FRAGMENT(
+    __contiguous_range_,
+    requires(_Tp& __t)(
+      requires(random_access_range<_Tp>),
+      requires(contiguous_iterator<iterator_t<_Tp>>),
+      requires(same_as<decltype(_CUDA_VRANGES::data(__t)), add_pointer_t<range_reference_t<_Tp>>>)
+    ));
+
+  template <class _Tp>
+  _LIBCUDACXX_CONCEPT contiguous_range = _LIBCUDACXX_FRAGMENT(__contiguous_range_, _Tp);
+
+  template <class _Tp>
+  _LIBCUDACXX_CONCEPT_FRAGMENT(
+    __common_range_,
+    requires()(
+      requires(range<_Tp>),
+      requires(same_as<iterator_t<_Tp>, sentinel_t<_Tp>>)
+    ));
+
+  template <class _Tp>
+  _LIBCUDACXX_CONCEPT common_range = _LIBCUDACXX_FRAGMENT(__common_range_, _Tp);
+
+  template <class _Tp>
+  _LIBCUDACXX_INLINE_VAR constexpr bool __is_std_initializer_list = false;
+
+  template <class _Ep>
+  _LIBCUDACXX_INLINE_VAR constexpr bool __is_std_initializer_list<initializer_list<_Ep>> = true;
```

Comment:
question: do we want `std::initializer_list` to be recognized as well?

Reply:
Actually `cuda::std::initializer_list` is just an alias for `::std::initializer_list` So we are already doing that